src/server/pfs/drive/driver.go

```diff
@@ -592,10 +713,35 @@ func filterBlockRefs(filterShard *pfs.Shard, blockRefs []*pfs.BlockRef) []*pfs.B
 	return result
 }
 
-func (d *driver) inspectFile(file *pfs.File, filterShard *pfs.Shard, shard uint64, from *pfs.Commit) (*pfs.FileInfo, []*pfs.BlockRef, error) {
+func (d *driver) getFileType(file *pfs.File, shard uint64) (pfs.FileType, error) {
```

Comment:
Is there a reason `inspectFile` can't be used in place of `getFileType` since the `pfs.FileInfo` it returns will have the file type set.

I guess it might be a little be less efficient but it seems like this is duplicating some code.